### PR TITLE
added tests to expansion

### PR DESCRIPTION
**Pull request by somnia-iuventi**

### CODE REVIEW

**Comment by nixpulvis** [comment]
AIs there a way we can write a helper function that creates a new faction/home system and creates it's initial `SystemFaction` linking the two. Ideally, this function could be used in both our tests here, and for the initial player setup.

I don't know the best way to write a helper function that either works with `Commands` or `World`, maybe there's a way to abstract over this?

File: src/system.rs [hunk 1]
```diff
@@ -1,5 +1,11 @@
 use bevy::prelude::*;
 
+#[derive(Bundle, Debug)]
+pub(crate) struct SystemBundle {
+    pub system: System,
+    pub position: Position,
+}
```
Comment (nixpulvis) [inline src/system.rs:7]: I'm not opposed to adding this. But I'm curious what value it adds over `spawn((System { ... }, Position(...))` since that will essentially do the same thing, no?

File: src/expansion.rs [hunk 2]
```diff
@@ -98,3 +98,175 @@ pub(crate) fn check_expansion(
...
+            system: System {
+                address: 0,
+                name: "System2".into(),
+            },
+            position: Position(Vec3::new(1., 1., 1.)),
+        });
+
+        // Run systems
+        app.update();
+        app.update();
```
Comment (nixpulvis) [inline src/expansion.rs:207]: Is updating twice really necessary? Why did expansion seem to work in `main` when update was only being hit once w/o `DefaultPlugins`, or as that actually running update multiple times as well?

**Review by nixpulvis — COMMENTED** [review]
Only other comment is that it might be nice to have separate tests for `check_expansion` and `expand` as well as a integration test for the two of them.

I'm not sure the best way to test that an event was sent, but that's what I'd like to test for `check_expansion`.

Then `expand` just needs to test that no duplicate system factions get spawned, and that the INF is all updated correctly.

File: src/system.rs (hunk 1 above)
Comment (nixpulvis) [reply src/system.rs:7]: I guess it's really just a uniform way to construct things until we get required components.

File: src/expansion.rs (hunk 2 above)
Comment (nixpulvis) [reply src/expansion.rs]: Nope, didn't seem to be.
